src/services/Store/BalanceService.tsx

```diff
@@ -27,9 +29,13 @@ export const convertBNToBigNumberJS = (bn: EthScanBN): BN => {
 };
 
 export const toBigNumberJS = (balances: EthScanBalanceMap): BalanceMap => {
-  return Object.fromEntries(
-    Object.keys(balances).map((key) => [key, convertBNToBigNumberJS(balances[key])])
-  );
+  return mapObjIndexed(convertBNToBigNumberJS, balances);
+};
+
+export const nestedToBigNumberJS = (
+  balances: EthScanBalanceMap<EthScanBalanceMap>
+): BalanceMap<BalanceMap> => {
+  return mapObjIndexed(toBigNumberJS, balances);
```

Comment:
Continuing: https://github.com/MyCryptoHQ/MyCrypto/pull/3664#discussion_r543267452_
This is precisely why we `bigify` in order to harmonise the types.
The function should be able to handle the conversion from any bigish type to bignumber.js.
The important part is that all our big numbers within the application are the type of the return value of `bigify`.